File: packages/types/src/models/audit.ts
export interface AuditLog {
    id: string;
    user_id: string | null;
    action: string;
    resource_type: string;
    resource_id: string | null;
    old_value: any | null;
    new_value: any | null;
    ip_address: string | null;
    user_agent: string | null;
    created_at: string;
    user: {
        id: string;
        full_name: string;
        email: string;
    } | null;
}

export interface AuditLogListResponse {
    items: AuditLog[];
    total: number;
    page: number;
    per_page: number;
}

export interface AuditLogFilters {
    page?: number;
    per_page?: number;
    user_id?: string;
    action?: string;
    resource_type?: string;
    start_date?: string;
    end_date?: string;
}

File: apps/api/src/schemas/audit.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from .auth import UserResponse


class AuditLogResponse(BaseModel):
    id: UUID
    user_id: UUID | None
    action: str
    resource_type: str
    resource_id: str | None
    old_value: dict[str, Any] | None
    new_value: dict[str, Any] | None
    ip_address: str | None
    user_agent: str | None
    created_at: datetime
    user: UserResponse | None = None

    model_config = ConfigDict(from_attributes=True)


class AuditLogListResponse(BaseModel):
    items: list[AuditLogResponse]
    total: int
    page: int
    per_page: int
